Common/Data/UniverseSelection/OptionChainUniverse.cs

```diff
@@ -30,23 +30,31 @@ namespace QuantConnect.Data.UniverseSelection
     /// </summary>
     public class OptionChainUniverse : Universe
     {
-        private static readonly IReadOnlyList<TickType> QuotesAndTrades = new[] { TickType.Quote, TickType.Trade };
+        private static readonly IReadOnlyList<TickType> dataTypes = new[] { TickType.Quote, TickType.Trade, TickType.OpenInterest };
 
         private BaseData _underlying;
         private readonly Option _option;
         private readonly UniverseSettings _universeSettings;
+        private SubscriptionManager _subscriptionManager;
+
+        private DateTime _cacheDate;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OptionChainUniverse"/> class
         /// </summary>
         /// <param name="option">The canonical option chain security</param>
         /// <param name="universeSettings">The universe settings to be used for new subscriptions</param>
+        /// <param name="subscriptionManager">The subscription manager used to return available data types</param>
         /// <param name="securityInitializer">The security initializer to use on newly created securities</param>
-        public OptionChainUniverse(Option option, UniverseSettings universeSettings, ISecurityInitializer securityInitializer = null)
+        public OptionChainUniverse(Option option, 
```

Comment:
But I think you are right - some classes could be 100% same, so will think how to reduce the duplication.